### ISSUE TITLE
CMake duplicates compiler flags

### ISSUE THREAD

**Issue by jeffhammond** [LLNL/RAJA#422]
I am using a compiler option that cannot be duplicated in the invocation thereof.  CMake duplicates my compiler flags and thus breaks this.

Do you have suggestions on how to solve this?  I found https://stackoverflow.com/questions/39758585/duplicate-compile-flag-in-cmake-cxx-flags but don't know how to apply it in RAJA.

Here is what it looks like:
```sh
git clean -dfx
cmake .. -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_C_COMPILER=$CC 
-DOpenMP_CXX_FLAGS=$MYFLAGS -DRAJA_ENABLE_OPENMP=On -DRAJA_ENABLE_TARGET_OPENMP=On
```
where `$MYFLAGS` contains an option `XYZ`.

This fails as follows:
```sh
make[2]: Entering directory '/home/jrhammon/Work/DOE/LLNL/RAJA/github/build'
[  1%] Building CXX object CMakeFiles/RAJA.dir/src/AlignedRangeIndexSetBuilders.cpp.o
$CXX   -DUSE_OPENMP -I/home/jrhammon/Work/DOE/LLNL/RAJA/github/include 
-I/home/jrhammon/Work/DOE/LLNL/RAJA/github/build/include  -Wall -Wextra 
$MYFLAGS  -std=c++17 -O3 -DNDEBUG -fPIC   $MYFLAGS  
-o CMakeFiles/RAJA.dir/src/AlignedRangeIndexSetBuilders.cpp.o 
-c /home/jrhammon/Work/DOE/LLNL/RAJA/github/src/AlignedRangeIndexSetBuilders.cpp
$CXX error: for the XYZ option: may only occur zero or one times!
```

I have substituted `$CC`, `$CXX`, `$MYFLAGS`, and `$XYZ` but I can share them via email if it helps.

**Comment by davidbeckingsale** [comment]
Hey @jeffhammond, have you tried applying the duplicate detection directly to the CMAKE_CXX_FLAGS variable?

I'll try and reproduce this locally, but email me the specific flags you are trying.

**Comment by davidbeckingsale** [comment]
Fixed by #423
